### ISSUE TITLE
[Bug]: Needing --no-half after 1.2.1 > 1.3.1 update

### ISSUE THREAD

**Issue by TijuanaKez** [lshqqytiger/stable-diffusion-webui-directml#151]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

After git pull to 1.3.1, suddenly all images are just a beige blur.

Tried with and without `Upcast cross attention layer to float32` , still beige.
Tried reverting Settings >Optimisations > to V1, still beige.

--no-half fixes it, but I didn't need this before on RX 6900 XT before updating, so this GPU should do half precision.

### Steps to reproduce the problem

Prompt Txt2Img, default settings, all extra extensions disabled.

### What should have happened?

Generate an image

### Commit where the problem happens

b6af0a3

### What Python version are you running on ?

Python 3.10.x

### What platforms do you use to access the UI ?

MacOS

### What device are you running WebUI on?

AMD GPUs (RX 6000 above)

### What browsers do you use to access the UI ?

Apple Safari

### Command Line Arguments

```Shell
--skip-torch-cuda-test --upcast-sampling --opt-split-attention
```

### List of extensions

Disabled

### Console logs

```Shell
Python 3.10.11 (main, Apr  7 2023, 07:31:31) [Clang 14.0.0 (clang-1400.0.29.202)]
Version: v1.3.1
Commit hash: b6af0a3809ea869fb180633f9affcae4b199ffcf
Installing requirements
Launching Web UI with arguments: --skip-torch-cuda-test --upcast-sampling --opt-split-attention
No module 'xformers'. Proceeding without it.
Warning: caught exception 'Torch not compiled with CUDA enabled', memory monitor disabled
*** "Disable all extensions" option was set, will only load built-in extensions ***
Loading weights [cc6cb27103] from /Users/Me/stable-diffusion-webui/models/Stable-diffusion/v1-5-pruned-emaonly.ckpt
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 4.3s (import torch: 1.0s, import gradio: 1.0s, import ldm: 0.4s, other imports: 0.7s, load scripts: 0.5s, create ui: 0.4s, gradio launch: 0.2s).
Creating model from config: /Users/Me/stable-diffusion-webui/configs/v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Applying optimization: Doggettx... done.
Textual inversion embeddings loaded(0): 
Model loaded in 4.6s (load weights from disk: 2.1s, create model: 0.4s, apply weights to model: 0.8s, apply half(): 0.6s, move model to device: 0.6s).
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:15<00:00,  1.89it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:14<00:00,  2.04it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:14<00:00,  2.11it/s]
```

**Comment by JeffreySmalley** [comment]
add --backend=directml to webui-user args and check back

**Comment by Tanuki33** [comment]
i personally using `--upcast-sampling` instead of `--no-half`.
I forgot since when i use that (maybe about 2 weeks), no problem so far.

**Comment by TijuanaKez** [comment]
> add --backend=directml to webui-user args and check back

```
Launching Web UI with arguments: --skip-torch-cuda-test --upcast-sampling --opt-split-attention --backend=directml
No module 'xformers'. Proceeding without it.
usage: launch.py [-h] [--update-all-extensions] [--skip-python-version-check] [--skip-torch-cuda-test]
                 [--reinstall-xformers] [--reinstall-torch] [--update-check] [--test-server]
                 [--skip-prepare-environment] [--skip-install] [--data-dir DATA_DIR] [--config CONFIG] [--ckpt CKPT]
                 [--ckpt-dir CKPT_DIR] [--vae-dir VAE_DIR] [--gfpgan-dir GFPGAN_DIR] [--gfpgan-model GFPGAN_MODEL]
                 [--no-half] [--no-half-vae] [--no-progressbar-hiding] [--max-batch-count MAX_BATCH_COUNT]
                 [--embeddings-dir EMBEDDINGS_DIR] [--textual-inversion-templates-dir TEXTUAL_INVERSION_TEMPLATES_DIR]
                 [--hypernetwork-dir HYPERNETWORK_DIR] [--localizations-dir LOCALIZATIONS_DIR] [--allow-code]
                 [--medvram] [--lowvram] [--lowram] [--always-batch-cond-uncond] [--unload-gfpgan]
                 [--precision {full,autocast}] [--upcast-sampling] [--share] [--ngrok NGROK]
                 [--ngrok-region NGROK_REGION] [--ngrok-options NGROK_OPTIONS] [--enable-insecure-extension-access]
                 [--codeformer-models-path CODEFORMER_MODELS_PATH] [--gfpgan-models-path GFPGAN_MODELS_PATH]
                 [--esrgan-models-path ESRGAN_MODELS_PATH] [--bsrgan-models-path BSRGAN_MODELS_PATH]
                 [--realesrgan-models-path REALESRGAN_MODELS_PATH] [--clip-models-path CLIP_MODELS_PATH] [--xformers]
                 [--force-enable-xformers] [--xformers-flash-attention] [--deepdanbooru] [--opt-split-attention]
                 [--opt-sub-quad-attention] [--sub-quad-q-chunk-size SUB_QUAD_Q_CHUNK_SIZE]
                 [--sub-quad-kv-chunk-size SUB_QUAD_KV_CHUNK_SIZE]
                 [--sub-quad-chunk-threshold SUB_QUAD_CHUNK_THRESHOLD] [--opt-split-attention-invokeai]
                 [--opt-split-attention-v1] [--opt-sdp-attention] [--opt-sdp-no-mem-attention]
                 [--disable-opt-split-attention] [--disable-nan-check] [--use-cpu USE_CPU [USE_CPU ...]] [--listen]
                 [--port PORT] [--show-negative-prompt] [--ui-config-file UI_CONFIG_FILE] [--hide-ui-dir-config]
                 [--freeze-settings] [--ui-settings-file UI_SETTINGS_FILE] [--gradio-debug] [--gradio-auth GRADIO_AUTH]
                 [--gradio-auth-path GRADIO_AUTH_PATH] [--gradio-img2img-tool GRADIO_IMG2IMG_TOOL]
                 [--gradio-inpaint-tool GRADIO_INPAINT_TOOL] [--gradio-allowed-path GRADIO_ALLOWED_PATH]
                 [--opt-channelslast] [--styles-file STYLES_FILE] [--autolaunch] [--theme THEME] [--use-textbox-seed]
                 [--disable-console-progressbars] [--enable-console-prompts] [--vae-path VAE_PATH]
                 [--disable-safe-unpickle] [--api] [--api-auth API_AUTH] [--api-log] [--nowebui] [--ui-debug-mode]
                 [--device-id DEVICE_ID] [--administrator] [--cors-allow-origins CORS_ALLOW_ORIGINS]
                 [--cors-allow-origins-regex CORS_ALLOW_ORIGINS_REGEX] [--tls-keyfile TLS_KEYFILE]
                 [--tls-certfile TLS_CERTFILE] [--disable-tls-verify] [--server-name SERVER_NAME] [--gradio-queue]
                 [--no-gradio-queue] [--skip-version-check] [--no-hashing] [--no-download-sd-model] [--subpath SUBPATH]
                 [--add-stop-route] [--controlnet-dir CONTROLNET_DIR]
                 [--controlnet-annotator-models-path CONTROLNET_ANNOTATOR_MODELS_PATH] [--no-half-controlnet]
                 [--controlnet-preprocessor-cache-size CONTROLNET_PREPROCESSOR_CACHE_SIZE]
                 [--controlnet-loglevel {DEBUG,INFO,WARNING,ERROR,CRITICAL}] [--ldsr-models-path LDSR_MODELS_PATH]
                 [--lora-dir LORA_DIR] [--scunet-models-path SCUNET_MODELS_PATH]
                 [--swinir-models-path SWINIR_MODELS_PATH]
launch.py: error: unrecognized arguments: --backend=directml
```

Am I forming that argument correctly?

> i personally using --upcast-sampling instead of --no-half.

That is in my arguments already, either by default, or perhaps from setting `Upcast cross attention layer to float32` . That used to be all that was needed, but something has changed since updating from 1.2.1 > 1.3.1

**Comment by JeffreySmalley** [comment]
Are you sure it is still the directml branch being pulled?

**Comment by TijuanaKez** [comment]
> Are you sure it is still the directml branch being pulled?

Gosh darn it, how did I wind up here? I didn't even notice I'd veered off the main branch github. I'm running on a Mac for starters 🤦. My apologies. 
I don't think I have the power to delete this, but maybe someone else can.

**Comment by JeffreySmalley** [comment]
No worries, easy to make mistakes like this. Asking got you back on track faster is all

**Comment by tobabot** [comment]
Hi, can I ask how you are running this on a mac with 6900xt ? my windows auto1111 wont work with it anymore

**Comment by TijuanaKez** [comment]
I was using just following the Apple hardware instructions which set it up for MPS.
It works, but MPS is just way behind in terms of performance to rocm, and there were to many quirks and glitches.
I resorted to installing Ubuntu on a spare drive and now its pretty close to CUDA speed.
I'm referring to stable diffusion in general. I take it you understand I posted this in the wrong GitHub page by mistake.
Directml=windows.
